# TP1 - Fecha de Entrega Jueves 21/05

# 7506 - Organizacion de Datos - TP1

# Trabajo Practico N°1

## Dataset `train`

# 1. Librerias

Importo las librerias que voy a utilizar para el Analisis Exploratorio

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# 2. Importo el Dataset

Importo los Dataset´s de los .csv individuales en DataFrames individuales

In [2]:
df_train = pd.read_csv("train.csv")

# 3. Caracteristicas Generales

In [3]:
#Dimension del DataFrame

df_train.shape

(7613, 5)

In [29]:
df_train.shape[0]

7613

In [4]:
#Tamaño del DataFrame (n° filas x n°columnas)

df_train.size

38065

In [5]:
#Columnas del DataFrame

df_train.columns

Index(['id', 'keyword', 'location', 'text', 'target'], dtype='object')

In [6]:
#Indice del DataFrame

df_train.index

RangeIndex(start=0, stop=7613, step=1)

In [7]:
#Observo las 5 primeras filas

df_train.head()

id keyword location                                               text  \
0   1     NaN      NaN  Our Deeds are the Reason of this #earthquake M...   
1   4     NaN      NaN             Forest fire near La Ronge Sask. Canada   
2   5     NaN      NaN  All residents asked to 'shelter in place' are ...   
3   6     NaN      NaN  13,000 people receive #wildfires evacuation or...   
4   7     NaN      NaN  Just got sent this photo from Ruby #Alaska as ...   

   target  
0       1  
1       1  
2       1  
3       1  
4       1

In [8]:
#Observo las 5 ultimas filas

df_train.tail()

id keyword location  \
7608  10869     NaN      NaN   
7609  10870     NaN      NaN   
7610  10871     NaN      NaN   
7611  10872     NaN      NaN   
7612  10873     NaN      NaN   

                                                   text  target  
7608  Two giant cranes holding a bridge collapse int...       1  
7609  @aria_ahrary @TheTawniest The out of control w...       1  
7610  M1.94 [01:04 UTC]?5km S of Volcano Hawaii. htt...       1  
7611  Police investigating after an e-bike collided ...       1  
7612  The Latest: More Homes Razed by Northern Calif...       1

In [10]:
#Resumen del DataFrame

df_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7613 entries, 0 to 7612
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   id        7613 non-null   int64 
 1   keyword   7552 non-null   object
 2   location  5080 non-null   object
 3   text      7613 non-null   object
 4   target    7613 non-null   int64 
dtypes: int64(2), object(3)
memory usage: 297.5+ KB


In [19]:
#Observo el tipo de datos por columna
tipos = df_train.dtypes
pd.DataFrame({'Columna':tipos.index,'Tipo de dato':tipos.values})

Columna Tipo de dato
0        id        int64
1   keyword       object
2  location       object
3      text       object
4    target        int64

Segun se observa se tienen 2 columnas con tipo de dato 'int64' y 3 columnas 'object'

In [11]:
#Verifico si alguna de las filas esta duplicada

df_train.duplicated().sum()

0

Segun se observa no existen filas duplicadas en su totalidad

## 3.1 Analisis de NaN 

In [25]:
#Cuento la cantidad de elementos NO NULOS de cada columna

cantidad_datos = df_train.count()
cantidad_datos

id          7613
keyword     7552
location    5080
text        7613
target      7613
dtype: int64

Realizando una comparacion con respecto al tamaño del DataFrame (7613 filas) se observa una diferencia en la cantidad de datos en las columnas "keyword" y "location", por lo tanto se analiza la existencia de NaN

In [26]:
#Evaluo por columna la existencia o no de NaN

tiene_nulos = df_train.isnull().any()

Como se identifico previamente en las columnas "keyword" y "location" existen valores NaN

In [34]:
#Cuento la cantidad de NaN por columna

nulos = df_train.isnull().sum()
pd.DataFrame({'Columna':nulos.index,'¿Tiene NaN?':tiene_nulos.values,'Cantidad de NaN':nulos.values,'% de NaN':((nulos.values*100/df_train.shape[0])).round(1)})

Columna  ¿Tiene NaN?  Cantidad de NaN  % de NaN
0        id        False                0       0.0
1   keyword         True               61       0.8
2  location         True             2533      33.3
3      text        False                0       0.0
4    target        False                0       0.0

Segun se observa tengo una cantidad importante de NaN en la columna "location"